extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/HttpBinderProcessor.java

```diff
@@ -110,12 +105,10 @@ void enableHttpServerSupport(Capabilities capabilities,
     }
 
     @BuildStep(onlyIf = HttpClientBinderEnabled.class)
-    void registerRestClientListener(BuildProducer<NativeImageResourceBuildItem> resource,
-            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
-        resource.produce(new NativeImageResourceBuildItem(
-                "META-INF/services/org.eclipse.microprofile.rest.client.spi.RestClientListener"));
-        reflectiveClass
-                .produce(new ReflectiveClassBuildItem(true, true, REST_CLIENT_METRICS_LISTENER));
+    void registerProvider(BuildProducer<AdditionalIndexedClassesBuildItem> additionalIndexed,
```

Comment:
This replaces a service loader listener responsible by registering the filters by a JAX-RS provider that implements the filters itself, right?
This looks much simpler to me but I fail to understand the priority that will be assigned to this provider...

Reply:
the priority assigned to the filters registered the other way is _exactly_ why they weren't called. (If you want a simple verification to check this, which I'm happy to have you try!, just remove the priority from the filter registration... )

This mechanism is exactly what the OTel RestClient support is using. I think making instantiation more similar between the two will make it easier to keep them aligned.

Reply:
ok... efective priority is still subject to research

Reply:
Clarified with @radcortez -- OpenTelemetry is no longer trying to use the templated URI for rest client traces, so we don't have an ordering concern between the two any more (as of #26694).